tests/app/celery/test_tasks.py

```diff
@@ -1021,6 +1021,31 @@ def test_save_letter_saves_letter_to_database_with_correct_postage(mocker, notif
     assert notification_db.postage == postage
 
 
+def test_save_letter_saves_letter_to_database_with_formatted_postcode(mocker, notify_db_session):
+    service = create_service(service_permissions=[LETTER_TYPE])
+    template = create_template(service=service, template_type=LETTER_TYPE)
+    letter_job = create_job(template=template)
+
+    mocker.patch('app.celery.tasks.letters_pdf_tasks.create_letters_pdf.apply_async')
+    notification_json = _notification_json(
+        template=letter_job.template,
+        to='Foo',
+        personalisation={'addressline1': 'Foo', 'addressline2': 'Bar', 'postcode': 'se1 64sa'},
```

Comment:
This PR is fine, but it might be night to have a test that illustrates the different formatting that might happen, for instance adding the space

Reply:
good point, we have lots of cases tested in utils, where that logic lives, I hope that is sufficient for now.